src/main/java/org/apache/mesos/scheduler/plan/DefaultBlock.java

```diff
@@ -0,0 +1,161 @@
+package org.apache.mesos.scheduler.plan;
+
+import org.apache.mesos.Protos;
+import org.apache.mesos.offer.OfferRequirement;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+/**
+ * This class is a default implementation of the Block interface.
+ */
+public class DefaultBlock implements Block {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final String name;
+    private final Optional<OfferRequirement> offerRequirementOptional;
+    private final UUID id = UUID.randomUUID();
+    private Status status;
+    private Map<Protos.TaskID, Status> tasks = new HashMap<>();
+
+    public DefaultBlock(String name, OfferRequirement offerRequirement, Status status) {
+        this(name, Optional.of(offerRequirement), status);
+    }
+
+    private DefaultBlock(String name, Optional<OfferRequirement> offerRequirementOptional, Status status) {
+        this.name = name;
+        this.offerRequirementOptional = offerRequirementOptional;
```

Comment:
IMO it's the compiler's job to keep track of types, no need to encode them in variable names.